### PR TITLE
Test skipping overhaul and documention

### LINKED ISSUE #23: Ensure steps after a failure are not executed
When a test is run and a step fails, subsequent steps still execute.

We cannot guarantee state so it just needs to not happen.

### PR DESCRIPTION
**Pull request by nfour**
Fixes #23 

## Changes

- Scenario steps now support `.skip` and the omission of the step fn:
  - `Scenario('foo').Then.skip('wew', myFn)` Skipped
  - `Scenario('foo').Then.skip('wew')` Skipped
  - `Scenario('foo').Then('wew')` Skipped
  - `Scenario('foo').Then('wew', myFn)` Not skipped!

- Scenario steps which error will no longer execute following steps
  - They will appear as a pass in Jest, but they are actually just skipped

- Steps are now prefixed with their step type name
- README refactor. Should be much better documented.

### CODE REVIEW

File: src/tests/execution/skippingStepTest.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Feature } from '../..';
+
+Feature('./execution.feature', ({ Scenario }) => {
+  Scenario('Steps are skipped when their function is omitted')
+    .Then('this test should be skipped')
+    .And.skip('this test should also be skipped');
```
Comment (adamelliottsweeting) [inline src/tests/execution/skippingStepTest.ts]: - The `.And` step should be skipped whether you add `.skip` or not, as the previous step was already skipped
- Ideally, you want a second scenario to show that using `.skip` also skips consecutive steps

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,15 @@ The format: [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 
 ## [Unreleased]
 
+- Scenario steps now support `.skip` and the omission of the step fn:
+  - `Scenario('foo').Then.skip('wew', myFn)` Skipped
+  - `Scenario('foo').Then.skip('wew')` Skipped
+  - `Scenario('foo').Then('wew')` Skipped
```
[the remark this answers is not in the document]
Comment (nfour) [reply CHANGELOG.md]: Just the one step. I'll remember to document this behaviour.

It gets replaced with an `identity` function (eg. lodash.identity)

File: src/tests/execution/__snapshots__/execution.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`Fermenter execution matching snapshots failingStepTest 1`] = `
+"FAIL src/tests/execution/failingStepTest.ts
+  Test Execution
+    Failing steps exit a scenario gracefully
+      ✓ Given: A step which passes
+      ✕ When: This step fails
+      ✓ Then: This step's function should not execute
```
[the remark this answers is not in the document]
Comment (nfour) [reply src/tests/execution/__snapshots__/execution.test.ts.snap]: So its jest that controls this, I can't change anything here without improvements to jest.

File: src/tests/execution/execution.feature [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Feature: Test Execution
+  Scenario: Failing steps exit a scenario gracefully
+    Given A step which passes
+    When This step fails
+    Then This step's function should not execute
```
[the remark this answers is not in the document]
Comment (nfour) [reply src/tests/execution/execution.feature]: <_<

File: src/tests/execution/skippingStepTest.ts (hunk 1 above)
Comment (nfour) [reply src/tests/execution/skippingStepTest.ts]: Great point, I started off just testing the compilation.

**Pull request merged**